### PR TITLE
extend AWS prowler v3 parser

### PR DESCRIPTION
**Pull request by kagahd**
Instead to add a new prowler v4 parser as suggested in PR #10338 this PR extends the existing prowler v3 parser to be able to import also prowler v4 reports in `json-ocsf` format.
This approach solves the deduplication problems between prowler v3 and v4 reports, described in PR #10338.

### CODE REVIEW

File: dojo/tools/aws_prowler_v3/parser.py
Comment (manuel-sommer) [inline dojo/tools/aws_prowler_v3/parser.py:23]: Could you please move the functionalities of v3 and v4 into separate files?

File: dojo/tools/aws_prowler_v3/parser.py
Comment (manuel-sommer) [inline dojo/tools/aws_prowler_v3/parser.py:20]: Please also reflect v4 in the SCAN_TYPE in lines 10,11,14,17 and rename this folder maybe from aws_prowler_v3 to aws_prowler_v3_v4
Also add the SCAN_TYPE to settings.dist.py deduplication

**Review by manuel-sommer — CHANGES_REQUESTED** [review]

File: dojo/tools/aws_prowler_v3/parser.py
Comment (kagahd) [reply dojo/tools/aws_prowler_v3/parser.py:20]: Hi @manuel-sommer, different scan_types would make it impossible to deduplicate formerly imported findings by prowler v3 with newly imported findings by prowler v4. Hence both parsers need to use the same scan_type.

File: dojo/tools/aws_prowler_v3/parser.py
Comment (kagahd) [reply dojo/tools/aws_prowler_v3/parser.py:23]: Why? What's the benefit in separating them? Other scanners do the same in the same manner.

Comment (Maffooch) [reply dojo/tools/aws_prowler_v3/parser.py:23]: Separate files for separate formats/extension is the standard the project is moving towards, largely thanks to @manuel-sommer

**Comment by mtesauro** [comment]
@kagahd Once the files are separated, this looks ready to approve.  Nice work so far :+1:

File: dojo/tools/aws_prowler_v3/parser.py
Comment (manuel-sommer) [reply dojo/tools/aws_prowler_v3/parser.py:20]: Hi @kagahd, you can rename the scantype from `AWS Prowler V3` to `AWS Prowler V3 and V4` just to reflect for the community that also V4 can be imported here. For exisitng findings, just add a migrations file, like https://github.com/DefectDojo/django-DefectDojo/pull/9355 or https://github.com/DefectDojo/django-DefectDojo/pull/9522. It basically migrates the exisiting findings to the new scantype name.

File: dojo/tools/aws_prowler_v3/parser.py
Comment (kagahd) [reply dojo/tools/aws_prowler_v3/parser.py]: Ok, I just separated them.

File: dojo/tools/aws_prowler_v3/parser.py
Comment (kagahd) [reply dojo/tools/aws_prowler_v3/parser.py]: Hi @manuel-sommer, I renamed the parser and its package to `AWSProwlerV3plus` to reflect for the community that besides of prowler v3 reports, also reports of later prowler versions can be imported here. 
Changing the `scan_type` accordingly would be fine but is out of my scope for this contribution because it requires too much time for me to implement which I currently don't have. For my company, it works as it is and I just wanted to share the work with the community without putting still more time in it.
It's a pity that the former PR for the prowler v3 parser was accepted. It would have been better to extend the existing prowler parser instead to introduce a new one which scan_type is/was bound to a specific prowler version.
Feel free to adapt the code to fit DefectDojo's requirements.

**Comment by kagahd** [comment]
As happened also in PR #10372, only 1 unit test in alpine and also in debian failed, but this is not due to my code contribution.
Related logs:
```
2024-06-11T08:19:36.4619889Z uwsgi-1  | FAIL: test_delete_preview (unittests.test_rest_framework.TestsTest.test_delete_preview)
2024-06-11T08:19:36.4620600Z uwsgi-1  | ----------------------------------------------------------------------
2024-06-11T08:19:36.4621097Z uwsgi-1  | Traceback (most recent call last):
2024-06-11T08:19:36.4621664Z uwsgi-1  |   File "/app/unittests/test_rest_framework.py", line 173, in wrapper
2024-06-11T08:19:36.4622208Z uwsgi-1  |     f(self, *args, **kwargs)
2024-06-11T08:19:36.4622976Z uwsgi-1  |   File "/app/unittests/test_rest_framework.py", line 544, in test_delete_preview
2024-06-11T08:19:36.4623853Z uwsgi-1  |     self.assertEqual(self.deleted_objects, len(response.data['results']), response.content[:1000])
2024-06-11T08:19:36.4628382Z uwsgi-1  | AssertionError: 5 != 18 : b'{"count":18,"next":null,"previous":null,"results":[{"model":"Test","id":3,"name":"ZAP Scan"},{"model":"Finding","id":3,"name":"High Impact Test Finding"},{"model":"Finding_found_by","id":3500,"name":"Finding_found_by object (3500)"},{"model":"Finding","id":2,"name":"High Impact Test Finding"},{"model":"Endpoint_Status","id":1,"name":"\'High Impact Test Finding\' on \'ftp://localhost\'"},{"model":"Finding_found_by","id":3499,"name":"Finding_found_by object (3499)"},{"model":"Finding","id":4,"name":"High Impact Test Finding"},{"model":"Finding_found_by","id":3501,"name":"Finding_found_by object (3501)"},{"model":"Finding","id":5,"name":"High Impact Test Finding"},{"model":"Finding_found_by","id":3502,"name":"Finding_found_by object (3502)"},{"model":"JIRA_Issue","id":2,"name":"Security How-to | Finding: High Impact Test Finding, ID: 5 | Jira Key: 222"},{"model":"Finding","id":6,"name":"High Impact Test Finding"},{"model":"Finding_found_by","id":3503,"name":"Finding_found_by object (3503)"},{'
2024-06-11T08:19:36.4632367Z uwsgi-1  | 
2024-06-11T08:19:36.4632761Z uwsgi-1  | ----------------------------------------------------------------------
2024-06-11T08:19:36.4633235Z uwsgi-1  | Ran 2693 tests in 338.437s
2024-06-11T08:19:36.4633558Z uwsgi-1  | 
2024-06-11T08:19:36.4633852Z uwsgi-1  | FAILED (failures=1, skipped=472)
```

**Comment by kagahd** [comment]
> @kagahd Once the files are separated, this looks ready to approve. Nice work so far 👍

Thanks @mtesauro! The files are separated now.

**Comment by mtesauro** [comment]
@kagahd 
Just re-kicked the REST tests - we've had issues with them lately.  There's a PR to fix this that you probably should wait till it's merged and rebase based on the dev branch after that PR is merged.

Those rest tests (prior to that PR) are super flaky and fail intermittently after moving them to Postgres in preparation for deprecating MySQL.

The PR that fixes it and to watch is #10387

**Comment by kagahd** [comment]
> you probably should wait till it's merged and rebase based on the dev branch after that PR is merged.

Hi @mtesauro , I followed your suggestion but now, the `Configured Codepaths Analyzer` is failing which was run successfully before the rebase on the dev branch.

**Comment by mtesauro** [comment]
About the Code Path Analyzer see:
https://github.com/DefectDojo/django-DefectDojo/pull/10384#issuecomment-2166000013

**Review by Maffooch — APPROVED** [review]

**Review by mtesauro — APPROVED** [review]
Approved

**Review by grendel513 — APPROVED** [review]

**Comment by kagahd** [comment]
It's more than **3 weeks ago** that I opened this pull request. If you don't merge it, merge conflicts will accumulate and will be harder to resolve. 
DefectDojo team, are you really **interested in contributions** or not?

**Review by hblankenship — APPROVED** [review]

**Comment by mtesauro** [comment]
@kagahd 
> It's more than **3 weeks ago** that I opened this pull request. If you don't merge it, merge conflicts will accumulate and will be harder to resolve. DefectDojo team, are you really **interested in contributions** or not?

Yes, we're always happy to have PRs from the community. It's summer time (at least in the northern hemisphere) and the combination of:
* several maintainers going on holiday recently
* several large PRs that were queued up for 2.36.0 release (which happened yesterday)
* preparation for deprecation of MySQL and RabbitMQ (see [here](https://github.com/DefectDojo/django-DefectDojo/discussions?discussions_q=author%3Amtesauro) especially [this one](https://github.com/DefectDojo/django-DefectDojo/discussions/9690))
* something strange happening with Github caches for GH Actions not being correctly evicted in a timely manner and slowing down our GH Action runs
* I was at OWASP's AppSec Lisbon presenting on DefectDojo last week so am just now catching up on PRs myself

has kept the maintainers busier that normal in June.

You've got the needed 4 approvals, if you can fix this one last merge conflict, I'm ready and very willing to merge this PR.

**Comment by kagahd** [comment]
Thanks for your explanations and willingness to merge the PR @mtesauro! 
It seems that all is green now, so it's ready to be merged 🎉

**Pull request merged**
